include/zio/zio_dev.h

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright (c) 2019 Thomas Burdick <thomas.burdick@gmail.com>
+ * Copyright (c) 2019 Kevin Townsend
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief ZIO device API definitions.
+ */
+
+#ifndef ZEPHYR_INCLUDE_ZIO_DEV_H_
+#define ZEPHYR_INCLUDE_ZIO_DEV_H_
+
+#include <zephyr/types.h>
+#include <device.h>
+
+#include <zio/zio_attr.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief ZIO device API definitions.
+ * @defgroup zio_device ZIO device definitions
+ * @ingroup zio
+ * @{
+ */
+
+struct zio_buf;
+
+/**
+ * A ZIO device provides attributes, channels, and triggers for a generalized IO
+ * device which can be described as a IO device with sampled input/output from
+ * one or more channels. Where the device and input/output channels may have a
+ * set of attributes themselves.
+ *
+ * Each device describes itself through a set of statically defined arrays
+ * accessible from this API
+ */
+struct zio_dev_chan {
+	/**
+	 * Name of the channel, may be null
+	 */
+	const char *name;
+
+	/**
+	 * Type of channel
+	 * many types are pre-described as an enum in zio_defs.h
+	 */
+	const u16_t type;
```

Comment:
Which enum? I see only attribute types, but this is a channel type. Could we use enum as a data type of this value?
Then there will be no doubts.

Reply:
We can use an enum, however the idea here is that the enum in zio_defs.h descibes known channel types, with the last item in the enum always being the limit of predefined types. This would allow additional enum types to be added for channel types, attributes, and others. I don't think thats clear in the current state of affairs, but thats the idea.

Reply:
Ok. Could we then state that in the comments? Which enum should be there and how to extend it?

Reply:
Yes of course!